client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/list.js

```diff
@@ -29,43 +29,89 @@ const renderRateNotice = translate => {
 	);
 };
 
-export const ShippingRates = ( {
-	id,
-	selectedRates, // Store owner selected rates, not customer
-	availableRates,
-	selectedPackages,
-	allPackages,
-	updateRate,
-	errors,
-	shouldShowRateNotice,
-	translate,
-} ) => {
-	const packageNames = getPackageDescriptions( selectedPackages, allPackages, true );
-	const hasSinglePackage = 1 === Object.keys( selectedPackages ).length;
-
-	const renderSinglePackage = ( pckg, pckgId ) => {
-		if ( ! ( pckgId in availableRates ) ) {
+const MAX_VISIBLE_RATE = 5; // Maximum number of rates to display in the shipping label modal
+
+class ShippingRates extends Component {
+	render() {
+		const packageNames = getPackageDescriptions( this.props.selectedPackages, this.props.allPackages, true );
+		const hasSinglePackage = 1 === Object.keys( this.props.selectedPackages ).length;
+
+		const shippingRates = Object.keys(this.props.selectedPackages).map((pckgId) => {
+			return <PackageShippingRates
+				key={pckgId}
+				id={this.props.id}
+				updateRate={this.props.updateRate}
+				translate={this.props.translate}
+				pckgId={pckgId}
+				selectedRate={this.props.selectedRates[ pckgId ] || ''}  // Store owner selected rates, not customer
+				availableRates={this.props.availableRates[ pckgId ]}
+				packageErrors = {this.props.errors[ pckgId ] || []}
+				packageNames = {packageNames[ pckgId ]}
+				hasSinglePackage = { hasSinglePackage }
+			/>
+		});
+		return <div>
+			{ this.props.shouldShowRateNotice && renderRateNotice( this.props.translate ) }
+			{ shippingRates }
+		</div>
+	}
+}
+
+class PackageShippingRates extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			packageRates: JSON.parse(JSON.stringify(this.props.availableRates.default.rates)),
+			visiblePackageRates: []
+		}
+
+		this.state.visiblePackageRates = this.state.visiblePackageRates.concat(this.state.packageRates.splice(0, MAX_VISIBLE_RATE))
```

Comment:
This feels very complicated.

Reply:
I needed a way to move the `packageRates` array elements from one end to the other. Something like this
```
[] [a, b, c, d]
[a] [b, c, d]
[a, b] c, d]
...
```
I have tried a few approaches with slice, splice, etc but none of the native JS function allows me to make this shorter. i am open to other ways.

Reply:
You store how much you need in a variable that you update whenever `Show more` is clicked. You use the value from that variable to update the number of things that you want to show.

You can `splice` or you can use `lodash` `take` for example: https://github.com/Automattic/woocommerce-services/pull/1782/commits/4aa43cd908554221401ea8694cecefc042a6c864#diff-8f6bc9adbfa669876e5a156a4c4411cdR53

Reply:
At least that is how I think about it. This way the number of values is stored explicitly.

Reply:
Since your version works I would not worry about this too much right now :P